onnxruntime/test/perftest/ort_test_session.cc

```diff
@@ -136,5 +136,68 @@ OnnxRuntimeTestSession::OnnxRuntimeTestSession(Ort::Env& env, std::random_device
   }
 }
 
+bool OnnxRuntimeTestSession::PopulateGeneratedInputTestData()
+{
+  size_t num_input_nodes;
+  num_input_nodes = session_.GetInputCount();
+
+  // iterate over all input nodes
+  for (size_t i = 0; i < num_input_nodes; i++) {
+    Ort::TypeInfo typeInfo = session_.GetInputTypeInfo(i);
+    Ort::MemoryInfo memory_info = Ort::MemoryInfo::CreateCpu(OrtArenaAllocator, OrtMemTypeDefault);
+    if (typeInfo.GetONNXType() == ONNX_TYPE_TENSOR) {
+        auto tensorInfo = typeInfo.GetTensorTypeAndShapeInfo();
+        std::vector<int64_t> input_node_dim = tensorInfo.GetShape();
+        int64_t input_tensor_size = 1;
+        for (auto dim : input_node_dim)
+        {
+          input_tensor_size *= dim;
+        }
+        size_t byteCount = input_tensor_size;
+        switch (tensorInfo.GetElementType())
+        {
+          case ONNX_TENSOR_ELEMENT_DATA_TYPE_FLOAT:
```

Comment:
Float16 and others are missing

Reply:
No need when using allocators. ORT will handle it internally with tensor_info.GetElementType().